src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java

```diff
@@ -190,7 +190,7 @@ public class JavadocTypeCheck
 
     /** Pattern to match type name within angle brackets in javadoc param tag. */
     private static final Pattern TYPE_NAME_IN_JAVADOC_TAG =
-            Pattern.compile("\\s*<([^>]+)>.*");
```

Comment:
The following removals weren't the cause of the issue, but are removed because they are not required.

### Removal of `\\s`:
We are guaranteed to have no space before type name, because the input is already trimmed in the following line (performed before this method is called):
https://github.com/checkstyle/checkstyle/blob/fa8b95c964251f2156db988c8d05215ea0a80368/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/utils/BlockTagUtil.java#L81

Call stack:
```java
extractBlockTags:57, BlockTagUtil (com.puppycrawl.tools.checkstyle.checks.javadoc.utils)
getJavadocTags:102,  JavadocUtil (com.puppycrawl.tools.checkstyle.utils)
getJavadocTags:374,  JavadocTypeCheck (com.puppycrawl.tools.checkstyle.checks.javadoc)
visitToken:323,      JavadocTypeCheck (com.puppycrawl.tools.checkstyle.checks.javadoc)
```
<hr>

### Removal of `>`:
If the type name starts with `<` it must be ended by `>` for it to be valid input, in any other case javadoc will throw an error and such tag wouldn't be rendered.
<hr>

### Removal of `>.*`:
What is beyond the type name does not concerns us in this particular method, hence no need to check anything.